Day1/Exercise2/solution.go

```diff
@@ -0,0 +1,40 @@
+package main
+
+import "fmt"
+
+type Node struct {
+	value string
+	left *Node
+	right *Node
+}
+
+
+
+func (root *Node) PreOrderTraversal() {
+	if root !=nil {
+		fmt.Println(root.value)
+		root.left.PreOrderTraversal()
+		root.right.PreOrderTraversal()
+	}
+
+}
+func (root *Node) PostOrderTraversal() {
+	if root !=nil {
+		root.left.PostOrderTraversal()
+		root.right.PostOrderTraversal()
+		fmt.Println(root.value)
+	}
+
+}
+func main(){
+	root := Node{"-",nil,nil}
+	root.left = &Node{"a",nil,nil}
+	root.right = &Node{"+",nil,nil}
+	root.right.left = &Node{"b",nil,nil}
+	root.right.right = &Node{"c",nil,nil}
+	fmt.Println("Preorder Traversal")
+	root.PreOrderTraversal()
```

Comment:
what is the output for this program? we should ideally do pre-order for the given input and post-order for the input, but your program takes input does pre-order and post-order is applied on the output of pre-order.